src/CattyTests/Embroidery/RunningStitchPatternTests.swift

```diff
@@ -0,0 +1,108 @@
+        embroideryStream!.activePattern!.spriteDidMove(to: CGPoint(x: 0, y: 80))
+
+        XCTAssertEqual(embroideryStream!.count, reference.count)
+        for i in 0..<reference.count {
+            XCTAssertEqual(reference[i].getPosition(), embroideryStream![i].getPosition())
+        }
+    }
+
+    func testSquare() {
+        let reference = EmbroideryStream()
+        reference.add(Stitch(x: 0, y: 0))
+        reference.add(Stitch(x: 0, y: 20))
+        reference.add(Stitch(x: 0, y: 40))
+        reference.add(Stitch(x: 0, y: 60))
+        reference.add(Stitch(x: 0, y: 80))
+        reference.add(Stitch(x: 20, y: 80))
+        reference.add(Stitch(x: 40, y: 80))
+        reference.add(Stitch(x: 60, y: 80))
+        reference.add(Stitch(x: 80, y: 80))
+        reference.add(Stitch(x: 60, y: 80))
+        reference.add(Stitch(x: 40, y: 80))
+        reference.add(Stitch(x: 20, y: 80))
+        reference.add(Stitch(x: 0, y: 80))
+        reference.add(Stitch(x: 0, y: 60))
+        reference.add(Stitch(x: 0, y: 40))
+        reference.add(Stitch(x: 0, y: 20))
+        reference.add(Stitch(x: 0, y: 0))
+
+        embroideryStream!.activePattern = RunningStitchPattern(for: embroideryStream!,
+                                                               at: .zero,
+                                                               with: 20)
+        embroideryStream!.activePattern!.spriteDidMove(to: CGPoint(x: 0, y: 80))
+        embroideryStream!.activePattern!.spriteDidMove(to: CGPoint(x: 80, y: 80))
+        embroideryStream!.activePattern!.spriteDidMove(to: CGPoint(x: 0, y: 80))
+        embroideryStream!.activePattern!.spriteDidMove(to: CGPoint(x: 0, y: 0))
+
+        XCTAssertEqual(embroideryStream!.count, reference.count)
+        for i in 0..<reference.count {
+            XCTAssertEqual(reference[i].getPosition(), embroideryStream![i].getPosition())
+        }
+    }
+
+    func testSquareNonDivisibleLength() {
+        let reference = EmbroideryStream()
+        reference.add(Stitch(x: 0, y: 0))
+        reference.add(Stitch(x: 0, y: 10))
+        reference.add(Stitch(x: 0, y: 19))
+        reference.add(Stitch(x: 10, y: 19))
+        reference.add(Stitch(x: 19, y: 19))
+        reference.add(Stitch(x: 19, y: 9))
+        reference.add(Stitch(x: 19, y: 0))
+        reference.add(Stitch(x: 9, y: 0))
+        reference.add(Stitch(x: 0, y: 0))
+
+        embroideryStream!.activePattern = RunningStitchPattern(for: embroideryStream!,
+                                                               at: .zero,
+                                                               with: 10)
+        embroideryStream!.activePattern!.spriteDidMove(to: CGPoint(x: 0, y: 19))
+        embroideryStream!.activePattern!.spriteDidMove(to: CGPoint(x: 19, y: 19))
+        embroideryStream!.activePattern!.spriteDidMove(to: CGPoint(x: 19, y: 0))
+        embroideryStream!.activePattern!.spriteDidMove(to: CGPoint(x: 0, y: 0))
```

Comment:
Sorry, I missed that one.
I think the asserts are missing in this tests right?

Reply:
You are right, totally missed that one, thank you.
I have updated the PR accordingly.